packages/lib/src/index.ts

```diff
@@ -168,7 +207,19 @@ export default function federation(
 
     transform(code: string, id: string) {
       for (const pluginHook of pluginList) {
-        const result = pluginHook.transform?.call(this, code, id)
+        const result = (
+          pluginHook.transform as ((
+            this: TransformPluginContext,
+            code: string,
+            id: string,
+            options?: {
+              ssr?: boolean
+            }
+          ) =>
+            | Promise<string | NullValue | Partial<SourceDescription>>
```

Comment:
Can this complex type be extracted and defined as a `type X = ...;` definition in the file header?